packages/devtools_app/lib/src/flutter/initializer.dart

```diff
@@ -99,19 +107,54 @@ class _InitializerState extends State<Initializer>
     );
 
     if (!connected) {
-      _navigateToConnectPage();
+      _handleNoConnection();
     }
   }
 
-  /// Goes to the connect page if the [service.serviceManager] is not currently connected.
-  void _navigateToConnectPage() {
+  /// Goes to the connect/disconnected page if the [service.serviceManager] is not currently connected.
+  void _handleNoConnection() {
     WidgetsBinding.instance.addPostFrameCallback((_) {
       if (!_checkLoaded() && ModalRoute.of(context).isCurrent) {
-        Navigator.of(context).popAndPushNamed('/');
+        Overlay.of(context).insert(_createDisconnectedOverlay());
       }
     });
   }
 
+  OverlayEntry _createDisconnectedOverlay() {
+    final theme = Theme.of(context);
+    OverlayEntry overlay;
+    overlay = OverlayEntry(
+      builder: (context) => Container(
+        color: const Color.fromRGBO(128, 128, 128, 0.5),
+        child: Center(
+          child: Column(
+            children: [
+              const Spacer(),
+              Text(
```

Comment:
There's quite a lot of fluff here, but I don't know if there's a terser way to stack the text/button and center within the screen.